compiler/sigmatch.nim

```diff
@@ -1089,8 +1089,13 @@ proc typeRelImpl(c: var TCandidate, f, aOrig: PType,
              else: isNone
 
   of tyAnything:
-    return if f.kind == tyAnything: isGeneric
-           else: isNone
+    if f.kind in {tyAnything}:
+      return isGeneric
+
+    if tfWildCard in a.flags and f.kind == tyTypeDesc:
```

Comment:
This is what ``tfWildCard`` says:


```nim

    tfWildcard # consider a proc like foo[T, I](x: Type[T, I])
                      # T and I here can bind to both typedesc and static types
                      # before this is determined, we'll consider them to be a
                      # wildcard type.
```

Which seems to be wrong. ``T`` and ``I`` cannot be ``typedesc`` types here, or maybe they cannot be "typedesc of typedesc". You can pass ``int`` to a generic, but not ``typedesc[int]``. In fact, this idea of turning the ``int`` in ``Container[int]`` into ``typedesc[int]`` is wrong.

Reply:
I'll think about it.

This particular block was changed while I was solving the numerous problems introduced by this patch. In the end, I've arrived at a solution that didn't need the particular changes here, so I removed them from the commit (they were introducing some code duplication). I've left only this modification, because it seemed innocent enough.